src/libs/actions/MergeTransaction.ts

```diff
@@ -191,7 +191,17 @@ function getOnyxTargetTransactionData(
         });
     }
 
-    return data.onyxData;
+    const onyxData = data.onyxData;
+
+    onyxData.optimisticData?.push({
```

Comment:
Can't we just do this?

```suggestion
    data.onyxData.optimisticData?.push({
```

Reply:
I prefer keep current implementation because we gonna add more in "Merge distance" PR